Document policy for python version support. (#5575)
quantumlib/Cirq: docs/start/install.md

To be informed of deprecations and breaking changes, subscribe to the
[cirq-announce google group mailing list](https://groups.google.com/forum/#!forum/cirq-announce).

## Python version supprt

Cirq currently supports python 3.7 and later.

> We follow numpy's schedule for python version support defined in [NEP 29](https://numpy.org/neps/nep-0029-deprecation_policy.html),

Reviewer: ```suggestion
We follow numpy's schedule for python version support defined in [NEP 29](https://numpy.org/neps/nep-0029-deprecation_policy.html);
```

Author: changed.

> though we may deviate from that schedule by extending support for older versions,

Reviewer: ```suggestion
though we may deviate from that schedule by extending support for older versions
```

Author: changed.

> if needed for compatibility with [Colab](https://colab.research.google.com/)

Reviewer: ```suggestion
if we need to be compatible with [Colab](https://colab.research.google.com/)
```

Author: reworded.
